packager/launchEditor.js

```diff
@@ -30,19 +50,22 @@ function launchEditor(fileName, lineNumber) {
     return;
   }
 
-  var argument = fileName;
+  var editor = process.env.REACT_EDITOR || process.env.EDITOR;
+  var args = [fileName];
   if (lineNumber) {
-    argument += ':' + lineNumber;
+    args = getArgumentsForLineNumber(editor, fileName, lineNumber);
   }
 
-  var editor = process.env.REACT_EDITOR || process.env.EDITOR;
   if (editor) {
     console.log('Opening ' + chalk.underline(fileName) + ' with ' + chalk.bold(editor));
-    exec(editor + ' ' + argument, function(error) {
-      if (error) {
-        console.log(chalk.red(error.message));
-        printInstructions('How to fix');
+
+    var child = spawn(editor, args, {stdio: 'inherit'});
```

Comment:
@pcottle cool - thanks for the clarification. `spawn` here sounds good to me then esp since some people were seeing nano silently eat up system resources. 

Can you handle the "error" event then? The docs say:
> Note that the exit-event may or may not fire after an error has occurred. If you are listening on both events to fire a function, remember to guard against calling your function twice.

https://iojs.org/api/child_process.html#child_process_event_error